jetson/percep/perception.hpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+#include <iostream>
+#include <opencv2/opencv.hpp>
+#include <opencv2/aruco.hpp>
+#include "config.h"
+#include "camera.hpp"
+#include "rover_msgs/Obstacle.hpp"
+#include <chrono>
+#include <cmath>
+#include <lcm/lcm-cpp.hpp>
+#include <sys/stat.h> // for disk writing
+#include <algorithm>
+#include <ctime>
+#include <string>
+#include <chrono>
+#include <thread>
+
+#if OBSTACLE_DETECTION
+/* --- PCL Includes --- */
+#include <pcl/filters/voxel_grid.h>
+#include <pcl/common/common_headers.h>
+#include <pcl/point_types.h>
+#include <pcl/filters/passthrough.h>
+#include <pcl/common/time.h>
+#include <pcl/segmentation/sac_segmentation.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/PointIndices.h>
+#include <pcl/ModelCoefficients.h>
+#include <pcl/kdtree/kdtree.h>
+#include <pcl/sample_consensus/method_types.h>
+#include <pcl/sample_consensus/model_types.h>
+#include <pcl/segmentation/extract_clusters.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/point_types.h>
+#if PERCEPTION_DEBUG
+  #include <pcl/visualization/pcl_visualizer.h>
+#endif
+#endif
+
+/* --- Ignore Error Checking for 3rd Part Header Files --- */
+#pragma GCC diagnostic ignored "-Wreorder"
+#pragma GCC diagnostic ignored "-Wcomment"
+
+#if ZED_SDK_PRESENT
+  #include <sl/Camera.hpp>
+#endif
+
+#if PERCEPTION_DEBUG
+  #include <opencv2/highgui/highgui.hpp>
+  #include <cstdlib>  
+#endif
+
+#pragma GCC diagnostic pop
+
+#define THRESHOLD_NO_WAY  80000 //how will we calibrate if the rover width changes
+//#define THRESHOLD_NO_OBSTACLE_CENTER  80000
+//#define THRESHOLD_NO_SUBWINDOW 27000
+#define THRESHOLD_CONFIDENCE 90
+#define SKY_START_ROW 400
+#define BALL_DETECTION_MAX_DIST 6.0  // this number is obtained from experiment. if the distance of the detected ball is greater than this number, false detection, we should ignore
+#define BALL_DETECTION_MIN_RAD 5
+#define BALL_DETECTION_MAX_RAD 300
```

Comment:
What do you think of moving these into config files like nav and localization have

Reply:
We are currently working on that